### DISCUSSION TITLE
How to get the onclick goal conversion tracking to work

### DISCUSSION THREAD

**Discussion by DominickV** [plausible/analytics#977] [Anything Else]
Hi, 

I am new to plausible and new to javascript but I am trying to get the goal conversion to work using an a href link and it seems fine and I followed the steps from the documentation but the dashboard doesnt seem to track it. 

I use typeform for lead generation and I want to track the buttons people click to open the form. 

I now have this link:
`       <a class="typeform-share link button -sm -outline-black text-black" href="XXX" data-mode="drawer_right" data-hide-headers=true target="_blank" onclick="plausible('Typeform', {props: {method: 'Feature film'}})">OPEN LEADGENERATION FORM </a> `

What am I doing wrong?

**Comment by Vigasaurus** [comment]
Your HTML there should work fine (I just tested it), you just need to make sure you've done all the steps as documented in the [custom event docs](https://plausible.io/docs/custom-event-goals). 

You need to: (the docs link above has these steps in order)
1. Ensure you have the second line after loading the Plausible script that makes the `window.plausible` object
2. Ensure that your custom event title is added as a valid event in your Plausible dashboard

**Reply by DominickV** [reply]
Thank you, I checked all steps but somehow I got 1 trigger triggered and the goal now shows 1 but I asked some friends to try it out and it never increased. This is the live page I am trying to fix: https://www.dominickverstoep.nl/trouwfilm-laten-maken.html

**Reply by Vigasaurus** [reply]
Ah okay, so what's happening there is that you have too many uncooperative events on a single tag - your `onclick` event for Plausible is working fine, the typeform integration, however, is setting a click event on the same tag with the function: 
```js
function(e) {
  return e.stopPropagation(), r.open(e), !1
}
```
and thus, it's overriding the Plausible onclick action, and stopping it from being sent. 

What you'll want to do instead of assigning your onclick event as a attribute on the <a> tag (something which Typeform can and does cancel), instead add it in a <script> tag **before** your Typeform embed script loads - this way, the click event you register will happen before the Typeform integration calls its click event which stops all other click events getting the event (that's what stopPropogation() does). Basically, you'll want to assign an ID or similar to any <a> to you want to be tracked, and in the script tag before Typeform's run
```js
document.getElementById('your-id').addEventListener('click', function() { plausible('your-event-target', {props: {your: 'props'}}) });
```

I'm not entirely certain that this will function the way I've outlined above/as intended, because I've never really worked with Typeform/its integration, but it theoretically *should*.

**Reply by DominickV** [reply]
I am somehow unable to get it to work. Do you have any recomendations?
